### PR TITLE
[@zeit/title_v3.4.x] Add definitions

### PR DESCRIPTION
**Pull request by pascalduez**
- Links to documentation: https://github.com/zeit/title
- Link to GitHub or NPM: https://github.com/zeit/title
- Type of contribution: new definition

### CODE REVIEW

File: definitions/npm/@zeit/title_v3.4.x/flow_v0.30.x-/title_v3.4.x.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+declare module '@zeit/title' {
+  declare type Options = $Exact<{
+    specials: Array<string>,
```
Comment (villesau) [inline definitions/npm/@zeit/title_v3.4.x/flow_v0.30.x-/title_v3.4.x.js]: Why $Exact and not `{||}`? Specials could perhaps be immutable `+specials`, as well as the type could be `$ReadOnlyArray`?

Comment (pascalduez) [reply definitions/npm/@zeit/title_v3.4.x/flow_v0.30.x-/title_v3.4.x.js]: > Why $Exact and not `{||}`
Not real reason, but I was thinking, isn't it a bit more "future proof" when they will remove the `{||}` notation?

Updating the rest, nice suggestions!

**Comment by pascalduez** [comment]
> Would perhaps be good to encourage some "best practices" regarding immutability, exactness etc.

I'm all for this.

Just wondering in this particular case whether it's that useful. Let's say one wants to reuse and mutate on options object. 

```js
let options = { specials: [''] };

title('', options);

options.specials.push(''); // Error

title('', options);
````

So we enforce immutability here, which is a good thing I guess.

```js
let options = { specials: [''] };

title('', options);

let newOptions = { specials: [...options.specials, ''] };

title('', newOptions);
```

File: definitions/npm/@zeit/title_v3.4.x/flow_v0.30.x-/title_v3.4.x.js (hunk 1 above)
Comment (villesau) [reply definitions/npm/@zeit/title_v3.4.x/flow_v0.30.x-/title_v3.4.x.js]: There will be codemod to get rid of it. It needs to be run against the whole repo neverthless. IT at least removes the pipes, not sure about $Exact, see https://github.com/flow-typed/flow-typed/issues/2876

**Comment by villesau** [comment]
It actually will not error, see: https://flow.org/try/#0PTAEAEDMBsHsHcBQiAmBTAxtAhgJzaJAK4B2GALgJawmjloDO5AFJSQA5HkBcoAJACU02FAHkS0AJ4BBXLmySAPCSIBbAEZpcAPgCUvAG6xKKANzIMNJqDy5QAXlABtALrnbAOk4MAFswCMAEwAzLrm9EzMtmFAA

Taking in immutable types ensures that the value is not mutated inside the function. If you take in mutable types, you can't pass immutable values to it: https://flow.org/try/#0PTAEAEDMBsHsHcBQiAmBTAxtAhgJzaJAK4B2GALgJawmjloDO5AFJSQA5HkBcoAgrlzYAngB4SRALYAjNLgB8ASl4A3WJRQBuZBhpNCtALyhmeXLwAkAJTTYUAeRLRhAoWIky5S0IfmgA3ogAkPRMpoKK2gC+iEA

**Comment by pascalduez** [comment]
My test where like: https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgDyOGAlnAHYDOYAvGACQCiAHgIYDGGAPAN6pgwAair4OpNjCoAuRgCU8bACaEKMLAEEATlrZZuVDFtIUA5gD4ANKgC+5gNzoleDjDZaCUAK4Uu5CmBkGDB4ABSGWrIRJqaWYHAk-lQA-LLEZJRUAJRRRjGOqCEY8YmZdGC8YKIuElKyANoA5I0AumA2BUEhoc1xCRnUWQX9SQB01eKSVKM4XlQAFj2NQ+hAA

**Comment by villesau** [comment]
I guess that's because Flow refines the type if no explicit type defined. If the type is explicitly defined, it does not error: https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgDyOGAlnAHYDOYAvGACQCiAHgIYDGGAPAN6pgwAair4OpNjCoAuRgCU8bACaEKMLAEEATlrZZuVDFtIUA5gD4ANKgC+5gNzoleDjDZaCUAK4Uu5CmBkGDB4ABSGWrIRJqaWYHAk-lQA-LLEZJRUAJRRRjGO2PhEiZkATHSMrJw8-IKiLhJSstq6+tFmVrYO6ByZGGBQAfShCRnUaSXUpVl05mC1gaTBYQDkK3GjSVmOgpuZAHT14pJU+zheVAAWoWvbtugPQA

**Pull request merged**